js/ui/workspace.js

```diff
@@ -1556,6 +1556,10 @@ const Workspace = new Lang.Class({
 
     _windowRemoved : function(metaWorkspace, metaWin) {
         this._doRemoveWindow(metaWin);
+
+        // Make sure to show the icon grid if we removed the last window
+        if (!Main.workspaceMonitor.hasActiveWindows)
+            Main.overview.showApps();
```

Comment:
I thought it would, but for some reason that does not work when you're in the "window picker" mode, since you apparently get the `destroy-completed` signal before the actual window has been removed, and `showApps()` never gets called, which is why I added this explicitly here.

I haven't investigated this deeply yet, but I'm guessing it might have something to do with the fact that when you close a window from the window picker you're not quite closing the real window at that time, but a clone of it. I can investigate it further if you want, but it looks like a safe workaround to me. What do you think?